fr/tri/Tri_par_insertion.md

```diff
@@ -34,12 +37,20 @@ Dans notre exemple, on va trier la liste $84, 33, 82, 53, 50, 66, 10, 92, 96, 77
 
 ### Exemple en vidéo
 
+Voici dans le cas général :
 ![Tri par insertion dans le cas général](../Exemples/.gif/TriInsertionGeneral.gif)
 
-### Conclusion
+Voici dans le meilleur cas :
+![Tri par insertion dans le meilleur cas](../Exemples/.gif/TriInsertionMeilleur.gif)
 
-On obtient à la fin la liste $10, 29, 33, 50, 53, 66, 77, 82, 84, 92, 96$ ce qui est bien notre liste de départ triée.
+Et voici dans le pire :
+![Tri par insertion dans le pire cas](../Exemples/.gif/TriInsertionPire.gif)
 
 ## Implémentation
 
 * [Python](https://github.com/TheAlgorithms/Python/blob/master/sorts/insertion_sort.py)
+
+## Pour aller plus loin
+
+* Dans le meilleur cas, nous n'effectuons qu'une unique comparaison. En effet, le terme comparé et plus grand que le terme juste avant, donc aucune modification n'est à effectuer. (ce qui correspond bien à $n$ opérations)
+* Dans le pire des cas, on doit au contraire "redescendre" tous les termes un par un. On effectue une opération pour remonter le 2ème terme, deux pour le 3ème, etc. On effectue donc au total $\sum{k=0}{n}{k}$ opération, soit bien un ordre de $N^2/2$ (ou plus précisément $N^2$)
```

Comment:
$N^2/2$ is the more precise term (which is $\Theta(N^2)$ OFC), obtained through applying the [Gaussian sum formula](https://en.wikipedia.org/wiki/Gauss_sum)

Reply:
yes, but we can't make a difference between the worst case and the average one if we only say the complexity is O(N^2)